### ISSUE TITLE
Memory leak problem with executePut similar to example

### ISSUE THREAD

**Issue by pkellner** [simoneb/axios-hooks#377]
My code is very similar to the code you have in your readme for executePut. The difference is I've consumed useAxios inside my own custom hook where I've exported my own `put` method.  I'm testing to see what happens if the component using `useAxios` is destroyed before `put` call completes and I was hoping that the `executePut would cancel but it does not seem to and I get the "memory leak from the browser" error.

Here is the code and the error.

warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.

```
const useSpeakerRequest = () => {
  const [{ data: getData, loading: getLoading, error: getError }] = useAxios(
    speakersRestUrl,
  );

  const [
    { data: putData, loading: putLoading, error: putError },
    executePut,
  ] = useAxios(
    {
      method: 'PUT',
    },
    { manual: true },
  );

  const put = (data,rollbackCallback,successCallback) => {
  
    const fetchData = async () => {
      await executePut({
        url: `${speakersRestUrl}/${data.id}`,
        data,
      });
      successCallback();
    }
    fetchData();
  };

  // if (getLoading || putLoading) return <p>Loading...</p>;
  // if (getError || putError) return <p>Error!</p>;

  return {
    data: getData,
    error: getError || putError,
    loading: getLoading || putLoading,
    put,
  };
};
```

**Comment by pkellner** [comment]
Just adding another comment about the issue.  I'm expecting that `successCallback` should never be hit when `await exectutePut` fails.  That is, I'm think what I want to happen is `executePut` fails and throws so that `successCallback` will not be called.

**Comment by simoneb** [comment]
Hi Peter, nice having you here, I used to follow your blog maybe ten years ago or so when I was doing .NET, this is an unexpected and very welcome visit.

I tried to put together a repro of the issue [here](https://codesandbox.io/s/axios-hooks-issue-377-repro-srno3?file=/src/index.js).

To the best of my understanding axios-hooks is behaving as intended and I have a feeling that the issue might be in your client code, which I can't see from the example you posted.

In the repro you can see that if you toggle the child (thereby forcing an unmount of the component that uses `axios-hooks`) while the data is being loaded with a manual request, you'll get an exception, which is right, but `axios-hooks` is not trying to update any state and that it handles its internal state correctly.

The exception you get is because when you do a manual request, `axios-hooks` expects you to handle errors coming back from the request.

Also note that if you unmount the child while the request is in progress you get an exception but not the "boom" exception, which you only get if you let the request complete. This demonstrates that if you unmount the component while the request is in progress the promise returned by the manual request function rejects, as it should.

In summary, I believe the issue you're seeing is in your client code and if you provide a full repro I can give it a look.

**Comment by pkellner** [comment]
Thanks @simoneb . You are correct in that it's my issue.  I went back and verified with axios by itself.  The solution is very thorny because I have competing interests in my real problem.  React makes it hard to optimize performance and at the same time deal with things like cancellations effects.  I've got this now and your repro helps.  Thank you :).  

BTW, I still do .net work (though don't blog much).  Since then, I've done 11 Pluralsight training courses, mostly around my new favorites, GraphQL and React.   Stay healthy as best you can.

**Comment by simoneb** [comment]
You're welcome, and congratulations for all the course work too.
